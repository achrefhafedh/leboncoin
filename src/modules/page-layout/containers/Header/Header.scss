@import '../../../../assets/colors.scss';

.header {
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
  background-color: $grey;
  li {
    cursor: pointer;
    list-style-type: none;
    display: inline-block;
    a,
    div {
      display: block;
      color: white;
      text-align: center;
      padding: 14px 16px;
      text-decoration: none;
      &:hover {
        background-color: $blue;
        color: white;
      }
      &.active {
        background-color: $blue;
      }
    }
    &:last-child {
      float: right;
    }
  }
}
