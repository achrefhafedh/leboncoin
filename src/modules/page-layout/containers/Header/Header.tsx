import React, { Component } from 'react';
import { NavLink, RouteComponentProps } from 'react-router-dom';
import { Redirect } from 'react-router-dom';

import { asyncLocalStorage } from 'modules/page-layout/helpers';

import './Header.scss';

type OwnProps = {};

type StateProps = {};

type DispatchProps = {};

type HeaderProps = StateProps & DispatchProps & OwnProps;

export type HeaderState = {
  redirect: boolean;
};

// a Class because it will be connected to store next time with other element
export class Header extends Component<HeaderProps, HeaderState> {
  constructor(props: HeaderProps) {
    super(props);
    this.state = {
      redirect: false,
    };
  }
  renderRedirect = () => {
    if (this.state.redirect) {
      return <Redirect to="/login" />;
    }
  };
  logout = () => {
    asyncLocalStorage.clear();
    this.setState({
      redirect: true,
    });
  };
  render() {
    return (
      <>
        {this.renderRedirect()}
        <ul className="header">
          <li>
            <NavLink exact to="/posts">
              Posts
            </NavLink>
          </li>
          <li>
            <NavLink exact to="/addPost">
              Add post
            </NavLink>
          </li>
          <li onClick={this.logout}>
            <div>Logout</div>
          </li>
        </ul>
      </>
    );
  }
}
