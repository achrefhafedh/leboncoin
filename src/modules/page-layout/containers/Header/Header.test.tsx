import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Header } from './Header';

configure({ adapter: new Adapter() });

describe('renders and execute function without crashing', () => {
  it('render without crashing', () => {
    const header = shallow(<Header />);
    expect(header).toMatchSnapshot();
  });

  it('execute function logout', () => {
    const wrapper = shallow<Header>(<Header />);
    wrapper.instance().logout();
  });
});
