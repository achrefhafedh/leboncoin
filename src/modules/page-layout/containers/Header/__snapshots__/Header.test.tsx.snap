// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and execute function without crashing render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        undefined,
        <ul
          className="header"
        >
          <li>
            <NavLink
              activeClassName="active"
              aria-current="page"
              exact={true}
              to="/posts"
            >
              Posts
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              aria-current="page"
              exact={true}
              to="/addPost"
            >
              Add post
            </NavLink>
          </li>
          <li
            onClick={[Function]}
          >
            <div>
              Logout
            </div>
          </li>
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                exact={true}
                to="/posts"
              >
                Posts
              </NavLink>
            </li>,
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                exact={true}
                to="/addPost"
              >
                Add post
              </NavLink>
            </li>,
            <li
              onClick={[Function]}
            >
              <div>
                Logout
              </div>
            </li>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                exact={true}
                to="/posts"
              >
                Posts
              </NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Posts",
                "exact": true,
                "to": "/posts",
              },
              "ref": null,
              "rendered": "Posts",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                exact={true}
                to="/addPost"
              >
                Add post
              </NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Add post",
                "exact": true,
                "to": "/addPost",
              },
              "ref": null,
              "rendered": "Add post",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                Logout
              </div>,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Logout",
              },
              "ref": null,
              "rendered": "Logout",
              "type": "div",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          undefined,
          <ul
            className="header"
          >
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                exact={true}
                to="/posts"
              >
                Posts
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                exact={true}
                to="/addPost"
              >
                Add post
              </NavLink>
            </li>
            <li
              onClick={[Function]}
            >
              <div>
                Logout
              </div>
            </li>
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  exact={true}
                  to="/posts"
                >
                  Posts
                </NavLink>
              </li>,
              <li>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  exact={true}
                  to="/addPost"
                >
                  Add post
                </NavLink>
              </li>,
              <li
                onClick={[Function]}
              >
                <div>
                  Logout
                </div>
              </li>,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  exact={true}
                  to="/posts"
                >
                  Posts
                </NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Posts",
                  "exact": true,
                  "to": "/posts",
                },
                "ref": null,
                "rendered": "Posts",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  exact={true}
                  to="/addPost"
                >
                  Add post
                </NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Add post",
                  "exact": true,
                  "to": "/addPost",
                },
                "ref": null,
                "rendered": "Add post",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  Logout
                </div>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Logout",
                },
                "ref": null,
                "rendered": "Logout",
                "type": "div",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
